datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -232,8 +232,61 @@ impl Hash for CreateExternalTable {
     }
 }
 
+impl PartialOrd for CreateExternalTable {
```

Comment:
Yeah, I'm not super happy with the way the syntax on this turned out, but the first field, `schema` prevents us from deriving it. I was looking at some alternatives, but it looks like Rust doesn't have a way to ignore a field for a derived trait by default. One option is to use something like the [derivative](https://crates.io/crates/derivative) crate, but that means another attribute for the struct, and additional attributes for the field.

In general, `DFSchemaRef` and `Schema` are the main blockers for deriving PartialOrd for structs, with the other cases (also) having HashMaps.

Reply:
For the ticket tracking thing, now that you mention it, it might be possible to give `DFSchemaRef` a manual implementation of `PartialOrd` that just gives back `None`, then everything else (almost) can derive it!